ui/app/AppLayouts/Chat/views/communities/JoinCommunityView.qml

```diff
@@ -0,0 +1,280 @@
+import QtQuick 2.14
+import QtQuick.Controls 2.14
+import QtQuick.Layouts 1.14
+import QtGraphicalEffects 1.0
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+import StatusQ.Controls 0.1
+
+import AppLayouts.Chat.panels.communities 1.0
+import AppLayouts.Chat.views 1.0
+
+import StatusQ.Layout 0.1
+
+import utils 1.0
+
+StatusSectionLayout {
+    id: root
+
+    // General properties:
+    property bool amISectionAdmin: false
+    property bool openCreateChat: false
+    property string name
+    property string communityDesc
+    property color color
+    property string channelName
+    property string channelDesc
+    property bool joinCommunity: true // Otherwise it means join channel action
+
+    // Permission overlay view properties:
+    property bool requirementsMet: true
+    property bool isInvitationPending: false
+    property bool isJoinRequestRejected: false
+    property var communityHoldings
+    property var viewOnlyHoldings
+    property var viewAndPostHoldings
+    property var moderateHoldings
+
+    // Blur view properties:
+    property int membersCount
+    property url image
+    property var communityItemsModel
+    property string chatDateTimeText
+    property string listUsersText
+    property var messagesModel
+
+    signal infoButtonClicked
+    signal adHocChatButtonClicked
+    signal revealAddressClicked
+    signal invitationPendingClicked
+
+    QtObject {
+        id: d
+
+        readonly property int blurryRadius: 32
```

Comment:
imo, while `blurryRadius` is used only once it could be inlined and `d` obj removed